lib/schema-fs-loader.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const path = require('path');
+const vm = require('vm');
+const metasync = require('metasync');
+const fs = require('fs');
+
+const { create } = require('./schema');
+const { all: decorators } = require('./decorators');
+
+const SCRIPT_TIMEOUT = 2000;
+const DOMAINS_SCHEMA = 'domains.schema';
+
+const moduleDir = path.dirname(__dirname);
+const schemasDir = path.join(moduleDir, 'schemas');
+
+const processSchemaFile = (filename, decorators, source, api = {}) => {
+  const sandbox = Object.assign({}, decorators, { api });
+  sandbox.global = sandbox;
+  const code = `'use strict';\n(${source})`;
+  const options = {
+    filename,
+    timeout: SCRIPT_TIMEOUT,
+    lineOffset: 1,
+  };
+  return vm.runInNewContext(code, sandbox, options);
+};
+
+// Load schema file from the filesystem
+//   filepath - <string> full path to the schema file
+//   api - <Object> object to be passed as 'api' during loading
+//   callback - <Function> to report the result
+//     error - <Error> error or null
+//     name - <string> schema name
```

Comment:
To be correctly parsed, there should be a comma after types.
```suggestion
//     name - <string>, schema name
```

Reply:
As I asked in a comment to the refactoring PR, could we avoid this (https://github.com/metarhia/common/pull/194#issuecomment-433872681)?